doc/sphinx/proofs/writing-proofs/inductive.rst

```diff
@@ -8,14 +8,14 @@ Applying constructors
 The tactics presented here specialize :tacn:`apply` (and
 :tacn:`eapply`) to the case of constructors of inductive types.
 
-.. tacn:: constructor @natural
-   :name: constructor
+.. tacn:: constructor {? @nat_or_var } {? with @bindings }
 
-   This tactic applies to a goal such that its conclusion is an inductive
-   type (say :g:`I`). The argument :token:`natural` must be less or equal to the
-   numbers of constructor(s) of :g:`I`. Let :n:`c__i` be the i-th
-   constructor of :g:`I`, then :g:`constructor i` is equivalent to
-   :n:`intros; apply c__i`.
+   .. todo PR: This really does an intros, peeling off foralls and -> ?
+      If @nat_or_var is not given, it just picks the right contructor?
+      (in which case, why would you ever bother to give @nat_or_var?).
+
+   When the goal conclusion is an inductive type `I`, this is equivalent
```

Comment:
Can you give me precise language for your informal description?

Reply:
I think there's some language describing the `red` tactic that can be reused.  Here is some rough but precise language describing the tactic: Repeat the following process: if the whnf of the goal is an inductive type, apply the given constructor; if it is a forall, run `intro`; otherwise, fail.  Said another way, if after running `repeat intro; hnf`, the goal is an inductive type then `apply` the ith constructor, otherwise fail.